src/extension.ts

```diff
@@ -1737,8 +1747,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
```

Comment:
you can combine these conditions since they're both expected to be truthy. Additionally, an early-return pattern is preferable here, since the majority of work is done in the "else" block:

```
if (explorerNode && await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
    refreshExplorer();
    return;
}

const currentNs = await kubectlUtils.currentNamespace(kubectl);
...
```